experiments/xls/stream_logging.py

```diff
@@ -21,4 +23,43 @@ def __exit__(self, exc_type, exc_value, traceback):
         xls_logger.removeHandler(self.handler)
 
     def get_log(self) -> str:
-        return self._stream.getvalue()
\ No newline at end of file
+        return self._stream.getvalue()
+
+
+class LogParser(object):
+    """
+    A parser that converts raw logs into dicts for ease of use
+    """
+
+    def __init__(self, raw_logs=[]):
+        self.raw_logs = raw_logs
+        self.logs = []
+        self.parse_logs(self.raw_logs)
+
+    def parse_logs(self, raw_logs):
+        for log in raw_logs:
+            self.parse(log)
+
+    def parse(self, raw_log):
+        if not raw_log:
+            return
+        log_tuple = raw_log.split('::')
+        if len(log_tuple) == 2:
+            self.add(log_tuple[0], log_tuple[1])
+        else:
+            self.add_error(raw_log)
+
+    def add(self, level, message):
```

Comment:
I suggest renaming this to "add_message", when I see LogParser.add in some other place, I need to check the signature to understand what it's doing, with renaming I don't.